docs/using.rst

```diff
@@ -188,6 +190,31 @@ expected output is::
 .. _README: ../components/centos/centos-k8s-singlenode-setup/README.rst
 .. _Vagrantfile: ../components/centos/centos-k8s-singlenode-setup/Vagrantfile
 
+Vagrant bi-directional folder sync
+==================================
+
+For basic usage please refer to the `Vagrant documentation. <https://www.vagrantup.com/docs/synced-folders/basic_usage.html>`_
+
+Vagrant's synced folders is a very powerful feature providing a simple way to move files (e.g code) between host and Vagrant guest.
+
+The following synced folder types work out of the box with the ADB Vagrant box, both for Virtualbox as well as Libvirt/KVM :
+
+* `vagrant-sshfs <https://github.com/dustymabe/vagrant-sshfs>`_ : works with Linux/GNU, OS X and Microsoft Windows.
+* `NFS <https://www.vagrantup.com/docs/synced-folders/nfs.html>`_ : works with Linux/GNU and OS X.
+
+There are also some other alternatives which, however, are currently not properly tested with ADB:
+
+* `SMB <https://www.vagrantup.com/docs/synced-folders/smb.html>`_ : For Microsoft Windows.
+
+  * You need to install cifs-utils RPM i.e. *sudo yum install cifs-utils* inside ADB for this to work.
+
+* `Virtualbox shared folder  <https://www.virtualbox.org/manual/ch04.html#sharedfolders>`_ : For Virtualbox users with Virtualbox guest additions.
+
+  * At this point of time Virtualbox guest additions do not come pre-installed in the ADB Vagrant box.
+  * Please refer to the upstream `Virtualbox documentation <https://www.virtualbox.org/manual/ch04.html>`_ to know about its installation steps.
+  * You can also use `vagrant-vbguest <https://github.com/dotless-de/vagrant-vbguest>`_ plugin to install Virtualbox guest additions in ADB Vagrant box.
```

Comment:
* Please refer to the upstream `Virtualbox documentation <https://www.virtualbox.org/manual/ch04.html>`_ to know about its installation steps.
Could be rewritten as For installation details please refer to  `Virtualbox documentation <https://www.virtualbox.org/manual/ch04.html>`_ 
Also this could be a part of the previous bullet point as it seems like a continuation of the point being made in the first bullet.